includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1174,25 +1187,31 @@ public function get_setup_intent_data_from_session( $payment_method = false ) {
 	 *
 	 * @param string $intent_id     The payment intent id.
 	 * @param string $client_secret The payment intent client secret.
+	 *
+	 * @return void
 	 */
 	private function add_upe_payment_intent_to_session( string $intent_id = '', string $client_secret = '' ) {
-		$cart_hash = 'undefined';
+		$woocommerce = WC();
+		if ( isset( $woocommerce->session ) ) {
+			$cart_hash = 'undefined';
+			if ( isset( $_COOKIE['woocommerce_cart_hash'] ) ) {
+				$cart_hash = sanitize_text_field( wp_unslash( $_COOKIE['woocommerce_cart_hash'] ) );
+			}
 
-		if ( isset( $_COOKIE['woocommerce_cart_hash'] ) ) {
-			$cart_hash = sanitize_text_field( wp_unslash( $_COOKIE['woocommerce_cart_hash'] ) );
+			$value = sprintf( '%s-%s-%s', $cart_hash, $intent_id, $client_secret );
+			$woocommerce->session->set( self::KEY_UPE_PAYMENT_INTENT, $value );
 		}
-
-		$value = $cart_hash . '-' . $intent_id . '-' . $client_secret;
-
-		WC()->session->set( self::KEY_UPE_PAYMENT_INTENT, $value );
 	}
 
 	/**
 	 * Removes the payment intent created for UPE from WC session.
+	 *
+	 * @return void
 	 */
 	public static function remove_upe_payment_intent_from_session() {
-		if ( isset( WC()->session ) ) {
-			WC()->session->__unset( self::KEY_UPE_PAYMENT_INTENT );
+		$woocommerce = WC();
+		if ( isset( $woocommerce->session ) && $woocommerce->session->__isset( self::KEY_UPE_PAYMENT_INTENT ) ) {
+			$woocommerce->session->__unset( self::KEY_UPE_PAYMENT_INTENT );
 		}
```

Comment:
```suggestion
		if ( isset( $woocommerce->session ) && isset( $woocommerce->session->self::KEY_UPE_PAYMENT_INTENT ) ) {
			unset( $woocommerce->session->self::KEY_UPE_PAYMENT_INTENT );
		}
```

I'm not sure about the actual semantics with using a `const` here, but the idea of magic methods is that we don't call them directly but allow PHP magic to do the trick 😉

Reply:
Better to use `set` with the `null` value like [the core](https://github.com/woocommerce/woocommerce/blob/733bcc88cf5e9186b1ae61eb89e30d4eeac74040/plugins/woocommerce/includes/class-wc-cart-session.php#L231-L239). I also think that makes sense as the default value when [trying to get a not-existing session key](https://github.com/woocommerce/woocommerce/blob/733bcc88cf5e9186b1ae61eb89e30d4eeac74040/plugins/woocommerce/includes/abstracts/abstract-wc-session.php#L101-L102) is `null`.

Reply:
Sounds good, addressed in 12aad6142388dd0946526e0ef29c4b7d2599c80a.